pkg/config/config.go

```diff
@@ -189,9 +245,33 @@ func (c *MicroshiftConfig) ReadFromConfigFile(configFile string) error {
 	if err != nil {
 		return fmt.Errorf("reading config file %q: %v", configFile, err)
 	}
+	var config Config
+	if err := yaml.Unmarshal(contents, config); err != nil {
+		return fmt.Errorf("decoding config file %s: %v", configFile, err)
+	}
 
-	if err := yaml.Unmarshal(contents, c); err != nil {
-		return fmt.Errorf("decoding config file %q: %v", configFile, err)
+	// Wire new Config type to existing MicroshiftConfig
```

Comment:
Is there a plan to use the new Config type directly in the future?

Reply:
Yes. The idea is to have a subsequent PR to remove MicroShift completely and only use Config moving forward. For now, the MicroShift remains to minimize the code change.